kv/tikv/storage/exec/scheduler.go

```diff
@@ -44,16 +50,30 @@ func (seq *Sequential) handleTask() {
 			continue
 		}
 
+		// Latch all keys required by the command. Make sure to call releaseLatches.
+		latches, err := task.cmd.WillWrite(reader)
+		if handleError(err, task, reader) {
+			continue
+		}
+		for {
+			wg := seq.acquireLatches(latches)
```

Comment:
no need to acquire locks for read only commands

Reply:
For readonly commands, latches will be empty and acquireLatches is a noop